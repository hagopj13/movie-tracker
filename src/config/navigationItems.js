import React from 'react';
import DiscoverIcon from '@material-ui/icons/Movie';
import UpcomingIcon from '@material-ui/icons/Update';
import ProfileIcon from '@material-ui/icons/Person';

export default [
  {
    id: 'DISCOVER',
    label: 'Discover',
    to: '/',
    icon: <DiscoverIcon />,
    requiresAuth: false,
  },
  {
    id: 'UPCOMING',
    label: 'Upcoming',
    to: '/upcoming',
    icon: <UpcomingIcon />,
    requiresAuth: false,
  },
  {
    id: 'PROFILE',
    label: 'Profile',
    to: '/profile',
    icon: <ProfileIcon />,
    requiresAuth: true,
  },
];
