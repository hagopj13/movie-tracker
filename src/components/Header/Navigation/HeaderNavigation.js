// @flow
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { NavLink } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';

import authSelectors from 'store/auth/auth.selectors';
import navigationItems from 'config/navigationItems';

type Props = {
  isAuth: boolean,
};

const useStyles = makeStyles((theme) => ({
  navLink: {
    color: theme.palette.common.white,
    marginRight: theme.spacing(3),
    '&:last-child': {
      marginRight: 0,
    },
  },
  selectedNavLink: {
    color: theme.palette.primary.main,
  },
}));

export const HeaderNavigation = (props: Props) => {
  const { isAuth } = props;

  const classes = useStyles();

  return (
    <div>
      {navigationItems.map(
        (navItem) =>
          (isAuth || !navItem.requiresAuth) && (
            <Link
              key={navItem.id}
              component={NavLink}
              exact
              to={navItem.to}
              className={classes.navLink}
              activeClassName={classes.selectedNavLink}
            >
              <Typography variant="h6" component="span">
                {navItem.label}
              </Typography>
            </Link>
          ),
      )}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  isAuth: authSelectors.selectIsAuth,
});

export default connect(mapStateToProps)(HeaderNavigation);
