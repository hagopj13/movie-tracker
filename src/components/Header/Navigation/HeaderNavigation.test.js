import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Link from '@material-ui/core/Link';
import Typography from '@material-ui/core/Typography';

import navigationItems from 'config/navigationItems';

import { HeaderNavigation } from './HeaderNavigation';

describe('HeaderNavigation component', () => {
  let shallow;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    const mockProps = {
      isAuth: true,
    };
    wrapper = shallow(<HeaderNavigation {...mockProps} />);
  });

  it('should render the HeaderNavigation component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Link for each navigation item', () => {
    expect(wrapper.find(Link)).toHaveLength(navigationItems.length);
  });

  it('should pass the path to the "to" prop of NavLink', () => {
    expect(wrapper.find(Link).at(0).prop('to')).toBe(navigationItems[0].to);
  });

  it('should pass the label to the "primary" prop of ListItemText', () => {
    expect(wrapper.find(Typography).at(0).text()).toBe(navigationItems[0].label);
  });

  it('should only render navigation items that do no require auth if isAuth is false', () => {
    const mockProps = {
      isAuth: false,
    };
    const newWrapper = shallow(<HeaderNavigation {...mockProps} />);
    const publicNavItems = navigationItems.filter((navItem) => !navItem.requiresAuth);
    expect(newWrapper.find(Link)).toHaveLength(publicNavItems.length);
  });
});
