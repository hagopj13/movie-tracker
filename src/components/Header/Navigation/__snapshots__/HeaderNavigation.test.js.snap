// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavigation component should render the HeaderNavigation component correctly 1`] = `
<div>
  <WithStyles(ForwardRef(Link))
    activeClassName="makeStyles-selectedNavLink-2"
    className="makeStyles-navLink-1"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "NavLink",
        "propTypes": Object {
          "activeClassName": [Function],
          "activeStyle": [Function],
          "aria-current": [Function],
          "className": [Function],
          "exact": [Function],
          "innerRef": [Function],
          "isActive": [Function],
          "location": [Function],
          "onClick": [Function],
          "replace": [Function],
          "sensitive": [Function],
          "strict": [Function],
          "style": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    exact={true}
    key="DISCOVER"
    to="/"
  >
    <WithStyles(ForwardRef(Typography))
      component="span"
      variant="h6"
    >
      Discover
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Link))>
  <WithStyles(ForwardRef(Link))
    activeClassName="makeStyles-selectedNavLink-2"
    className="makeStyles-navLink-1"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "NavLink",
        "propTypes": Object {
          "activeClassName": [Function],
          "activeStyle": [Function],
          "aria-current": [Function],
          "className": [Function],
          "exact": [Function],
          "innerRef": [Function],
          "isActive": [Function],
          "location": [Function],
          "onClick": [Function],
          "replace": [Function],
          "sensitive": [Function],
          "strict": [Function],
          "style": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    exact={true}
    key="UPCOMING"
    to="/upcoming"
  >
    <WithStyles(ForwardRef(Typography))
      component="span"
      variant="h6"
    >
      Upcoming
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Link))>
  <WithStyles(ForwardRef(Link))
    activeClassName="makeStyles-selectedNavLink-2"
    className="makeStyles-navLink-1"
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "NavLink",
        "propTypes": Object {
          "activeClassName": [Function],
          "activeStyle": [Function],
          "aria-current": [Function],
          "className": [Function],
          "exact": [Function],
          "innerRef": [Function],
          "isActive": [Function],
          "location": [Function],
          "onClick": [Function],
          "replace": [Function],
          "sensitive": [Function],
          "strict": [Function],
          "style": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    exact={true}
    key="PROFILE"
    to="/profile"
  >
    <WithStyles(ForwardRef(Typography))
      component="span"
      variant="h6"
    >
      Profile
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Link))>
</div>
`;
